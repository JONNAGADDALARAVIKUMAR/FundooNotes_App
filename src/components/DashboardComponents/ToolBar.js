import React, { Component } from 'react';
import {View, TouchableOpacity} from 'react-native';
import strings from '../../language/Languages';
import {Appbar, Avatar, Searchbar} from 'react-native-paper';
import DashBoardScreenStyles from '../../styles/DashBoardScreenStyles';

export default class ToolBar extends Component {
    constructor() {
        super();
        this.state = {
            
        }
    }
    render() {
        return(
            <View>
                <Appbar style = {DashBoardScreenStyles.App_Bar_Style}>
                        <Appbar.Action
                            icon="menu"
                            onPress={() => console.log('Pressed Menu')}
                        />
                        <Searchbar
                            style ={DashBoardScreenStyles.Search_Bar_Style}
                            placeholder = {strings.Search}
                            onChangeText = {data => console.log(data)}
                        />
                        <Appbar.Action icon = { (this.state.changeLayout ) ? "view-agenda-outline" : "view-grid-outline"} onPress  ={() => {
                            if(this.state.changeLayout) {
                                this.setState({
                                changeLayout: false
                            })} else {
                                this.setState({
                                    changeLayout: true
                                })
                            }
                            console.log('Layout Changed')}}
                        />
                        <TouchableOpacity onPress = {() => console.log('Pressed Profile')}>
                            <Avatar.Image
                                size = {35}
                                source = {require("../../assets/profilePic.jpg")}
                                style = {DashBoardScreenStyles.Profile_Style}
                            />
                        </TouchableOpacity>
                    </Appbar>
            </View>
        )
    }
}