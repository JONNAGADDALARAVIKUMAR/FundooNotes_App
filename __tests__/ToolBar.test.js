import React from 'react';
import {shallow} from 'enzyme';
import ToolBar from '../src/components/DashboardComponents/ToolBar';
import {Appbar} from 'react-native-paper';

describe('test ToolBar Screen', () => {
    it('test when render should match to snapshot', async () => {
        const component = shallow(<ToolBar/>)
        expect(component).toMatchSnapshot();
    })
    it('test the changeLayout method on Press Event should Update State', async () => {
        const onPressEvent = jest.fn();
        const component = shallow(<ToolBar onPress = {onPressEvent}/>)
        const instance = component.instance();
        instance.setState({
            changeLayout: false
        })
        instance.changeLayout();
        expect(onPressEvent).toHaveBeenCalled();
        expect(onPressEvent).toHaveBeenCalledTimes(1);
        expect(component.find(Appbar.Action).at(1).props().icon).toEqual('view-agenda-outline')
        expect(instance.state.changeLayout).toBe(true)

        instance.changeLayout();
        expect(onPressEvent).toHaveBeenCalled();
        expect(onPressEvent).toHaveBeenCalledTimes(2);
        expect(component.find(Appbar.Action).at(1).props().icon).toEqual('view-grid-outline')
        expect(instance.state.changeLayout).toBe(false)
    })
})