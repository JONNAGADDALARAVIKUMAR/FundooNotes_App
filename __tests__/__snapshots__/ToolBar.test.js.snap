// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ToolBar Screen test when render should match to snapshot 1`] = `ShallowWrapper {}`;
